src/app.rs

```diff
@@ -153,7 +183,9 @@ impl App{
                         });
                     }
                     ctx.layer();
-                    for elem in App::calculate_footprint(tf) {
+                    for elem in App::calculate_footprint(
```

Comment:
We are generating a whole bunch of tf lookups here

Reply:
why? we are iterating on a vector! I doubt calculate_footprint is called at every iteration of the for... Riiiight?

Reply:
you are right, my bad